projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts

```diff
@@ -72,6 +72,54 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
     return this._operationLogsService.isLogVisible('documents');
   }
 
+  /** Source for imported record. */
+  get source() {
+    if (this._activatedRouter.snapshot && this._activatedRouter.snapshot.params && this._activatedRouter.snapshot.params.type !== null) {
+      return this._activatedRouter.snapshot.params.type.replace('import_', '');
+    }
+    return null;
+  }
+
+  /** External identifier for imported record. */
+  get pid(): string | null {
+    if (this._activatedRouter.snapshot && this._activatedRouter.snapshot.params && this._activatedRouter.snapshot.params.pid !== null) {
+      return this._activatedRouter.snapshot.params.pid;
+    }
+    return null;
+  }
```

Comment:
Suggestion from google
```typescript
constructor(private route: ActivatedRoute) { }

  ngOnInit() {
    this.route.queryParams
      .subscribe(params => {
        console.log(params); // { orderby: "price" }
        this.orderby = params.orderby;
        console.log(this.orderby); // price
      }
    );
  }
```
it avoids a lot of conditions and store the param into a component variable (if use at multiple place)

Reply:
I just reorganized the code. I didn't touch this part. This can be refactored if you want, but I'm not going to do that again now.